ml-agents/mlagents/trainers/torch/utils.py

```diff
@@ -158,18 +205,37 @@ def create_input_processors(
         :param unnormalized_inputs: Vector inputs that should not be normalized, and added to the vector
             obs.
         :param normalize: Normalize all vector inputs.
-        :return: Tuple of visual encoders and vector encoders each as a list.
+        :return: Tuple of :
+         - ModuleList of the encoders (None if the input requires to be processed with a variable length
+         observation encoder)
+         - A list of embedding sizes (0 if the input requires to be processed with a variable length
+         observation encoder)
+         - A list of the inputs that need to be processed by a variable length observation encder.
         """
         encoders: List[nn.Module] = []
         embedding_sizes: List[int] = []
-        for sen_spec in sensor_specs:
-            encoder, embedding_size = ModelUtils.get_encoder_for_obs(
-                sen_spec.shape, normalize, h_size, vis_encode_type
-            )
-            encoders.append(encoder)
-            embedding_sizes.append(embedding_size)
+        var_len_indices: List[int] = []
+        for idx, sen_spec in enumerate(sensor_specs):
+            if ModelUtils.can_encode_attention(sen_spec):
+                # This is a 2D tensor
+                # TODO : better if condition
+                var_len_indices.append(idx)
+                encoders.append(None)
```

Comment:
Should we append EntityEmbeddings here and move that class into utils.py?

Reply:
EntityEmbeddings takes only one input (the list of variable length tensors). It might be problematic if we have more than one BufferSensor on C# since we will have a list with more than one element. Do you have a suggestion to make it work ?

Reply:
good point